### ISSUE TITLE
Support for Smart Inverted Colors in macOS

### ISSUE THREAD

**Issue by axelvugts** [lhc70000/iina#1737]
**System and IINA version:**

- macOS 10.13.4
- IINA 1.0.0-beta2 Build 80

**Expected behavior:**
I'd like to see support for the new Smart Inverted Colors feature of macOS 10.13.3. This would allow for watching video content without having to disable "inverted colors". Videoplayback in QuickTime and Safari already behaves this way.

**Comment by alejx** [comment]
Good to see this feature landed in macOS. However, after a quick search in [Apple's SDK release notes](https://developer.apple.com/library/content/releasenotes/General/RN-macOS-10.13.4/index.html), I didn't find any new API about this. This might be a new API that will be released on the following releases. We'll keep an eye on this!

**Comment by saagarjha** [comment]
Sorry, did you end up finding API for this @alejx?

**Comment by alejx** [comment]
@saagarjha I didn't, but I found that invert color was not introduced to macOS recently.

**Comment by saagarjha** [comment]
No, this feature exists. Severely limited in implementation, but it's there. It's the same checkbox as invert used to be, but it behaves differently: try turning it on and watching [a video](https://www.apple.com/105/media/us/iphone-x/2017/01df5b43-28e4-4848-bf20-490c34a926a7/films/feature/iphone-x-feature-tpl-cc-us-20170912_1920x1080h.mp4) in Safari.

**Comment by saagarjha** [comment]
After a bit of searching, this appears to be a WebKit-only feature. This isn't something that we can support yet, at least until it comes to Cocoa proper.

**Comment by LeoNatan** [comment]
This is now supported with new APIs and asset catalogs.

**Comment by Akemi** [comment]
if upstream is meant as mpv then we can't do much about it. this is something the client user should do.

it seems like NSView doesn't have an equivalent of [accessibilityIgnoresInvertColors](https://developer.apple.com/documentation/uikit/uiview/2865843-accessibilityignoresinvertcolors) from UIView, which is kinda a pain. i tested it wirk safari and quicktime, whereas only the former works as described. looking at the webkit source it just listens to an accessibility option notification and adds an invert css filter to all video/image tags.

i played a bit around and could do something desirable:
```
NSWorkspace.shared().notificationCenter.addObserver(
            self,
            selector: #selector(self.invertNotification(_:)),
            name: NSNotification.Name.NSWorkspaceAccessibilityDisplayOptionsDidChange,
            object: NSWorkspace.shared())

@objc private func invertNotification(_ notification: Notification?) {
    let work = notification!.object as! NSWorkspace
    if #available(macOS 10.12, *) {
        if work.accessibilityDisplayShouldInvertColors == true {
            //add filter via --glsl-shaders=invert.hook
        } else {
            //remove filter
        }
    }
}
```
invert.hook
```
//!HOOK OUTPUT
//!BIND HOOKED

vec4 hook()
{
    return (1 - HOOKED_texOff(0));
}
```

i also tried to use a CIFilter and add it to the filters property of the layer. this didn't have the desired effect since (assuming here) i couldn't set the proper colorspace of the CIContext (isn't accessible in the layer?). 
CIFilter result:
![screenshot 2018-12-24 at 17 08 59](https://user-images.githubusercontent.com/680386/50405339-7da4bf00-07b3-11e9-9be9-02d5937ecf92.png)
desired result:
![screenshot 2018-12-24 at 17 09 33](https://user-images.githubusercontent.com/680386/50405340-7da4bf00-07b3-11e9-911b-af5c1b31051a.png)
